libsplinter/src/biome/oauth/store/mod.rs

```diff
@@ -0,0 +1,228 @@
+    }
+
+    /// Convert this OAuthUser into an update builder.
+    pub fn into_update_builder(self) -> OAuthUserUpdateBuilder {
+        let Self {
+            user_id,
+            email,
+            access_token,
+            refresh_token,
+            provider,
+        } = self;
+        OAuthUserUpdateBuilder {
+            user_id,
+            email,
+            access_token,
+            refresh_token,
+            provider,
+        }
+    }
+}
+
+/// Builder for OAuthUser structs
+#[derive(Default)]
+pub struct OAuthUserBuilder {
+    user_id: Option<String>,
+    email: Option<String>,
+
+    access_token: Option<String>,
+    refresh_token: Option<String>,
+    provider: Option<OAuthProvider>,
+}
+
+impl OAuthUserBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn with_user_id(mut self, user_id: String) -> Self {
+        self.user_id = Some(user_id);
+
+        self
+    }
+
+    pub fn with_email(mut self, email: String) -> Self {
+        self.email = Some(email);
+
+        self
+    }
+
+    pub fn with_access_token(mut self, access_token: String) -> Self {
+        self.access_token = Some(access_token);
+
+        self
+    }
+
+    pub fn with_refresh_token(mut self, refresh_token: String) -> Self {
+        self.refresh_token = Some(refresh_token);
+
+        self
+    }
+
+    pub fn with_provider(mut self, provider: OAuthProvider) -> Self {
+        self.provider = Some(provider);
+
+        self
+    }
+
+    /// Build an OAuthUser
+    pub fn build(self) -> Result<OAuthUser, InvalidStateError> {
+        Ok(OAuthUser {
+            user_id: self.user_id.ok_or_else(|| {
+                InvalidStateError("A user id is required to successfully build an OAuthUser".into())
+            })?,
+            email: self.email.ok_or_else(|| {
+                InvalidStateError("An email is required to successfully build an OAuthUser".into())
+            })?,
+            access_token: self.access_token.ok_or_else(|| {
+                InvalidStateError(
+                    "An access token is required to successfully build an OAuthUser".into(),
+                )
+            })?,
+            refresh_token: self.refresh_token,
+            provider: self.provider.ok_or_else(|| {
+                InvalidStateError(
+                    "An provider id is required to successfully build an OAuthUser".into(),
+                )
+            })?,
+        })
+    }
+}
+
+/// Builds an updated `OAuthUser` struct.
+///
+/// This builder only allows changes to the fields on an OAuthUser that may be
+/// updated.
+pub struct OAuthUserUpdateBuilder {
+    // "immutable" items
+    user_id: String,
+    email: String,
+    provider: OAuthProvider,
+
+    // "mutable" items
+    access_token: String,
+    refresh_token: Option<String>,
+}
+
+impl OAuthUserUpdateBuilder {
+    pub fn with_access_token(mut self, access_token: String) -> Self {
+        // contents omitted for brevity
```

Comment:
I assume this comment is copy pasta (same in method below).

Reply:
Ha, yes

Reply:
Removed, as well as documented the functions.